### PR TITLE
advisor: Extend `Vulnerability` by the properties `summary` and `description`

### PR DESCRIPTION
**Pull request by fviernau**
Add the properties and implement the mapping for all advise providers where applicable.
See individual commits.

Context: #5593.

### CODE REVIEW

**Comment by fviernau** [comment]
@oheger-bosch due to the revert commit, your approval should be mendatory to merge this as per our discussion about Bosch's internal setup.

File: model/src/main/kotlin/Vulnerability.kt [hunk 1]
```diff
@@ -31,6 +34,18 @@ data class Vulnerability(
      */
     val id: String,
 
+    /**
+     * A short description of this [Vulnerability].
```
Comment (sschuberth) [inline model/src/main/kotlin/Vulnerability.kt:38]: I believe we shouldn't repeat the word "description" here to differentiate from the below. Just say "short summary".

File: model/src/main/kotlin/Vulnerability.kt [hunk 2]
```diff
@@ -31,6 +34,18 @@ data class Vulnerability(
...
+     */
+    @JsonInclude(Include.NON_NULL)
+    val summary: String? = null,
+
+    /**
+     * A description of this [Vulnerability].
```
Comment (sschuberth) [inline model/src/main/kotlin/Vulnerability.kt:44]: To complement the above, I'd write "full description" here.

File: reporter/src/funTest/kotlin/TestData.kt [hunk 3]
```diff
@@ -452,8 +452,8 @@ val ORT_RESULT = OrtResult(
 )
 
 val VULNERABILITY = Vulnerability(
-    "CVE-2021-1234",
-    listOf(
+    id = "CVE-2021-1234",
```
Comment (oheger-bosch) [inline reporter/src/funTest/kotlin/TestData.kt:455]: Would it make sense to add the new attributes to the test data?

Comment (fviernau) [reply reporter/src/funTest/kotlin/TestData.kt:455]: This test data is only used for testing the evaluator currently. 
I wouldn't want to add a test to the evaluator which covers `summary` or `description` as I believe these attributes are irrelevant for the rules, e.g. they are free text. Without adding a test case to the evaluator I believe it doesn't make sense to add to the test data. Are you ok with leaving it as-is?

Comment (oheger-bosch) [reply reporter/src/funTest/kotlin/TestData.kt:455]: Ok.

**Review by oheger-bosch — APPROVED** [review]

**Pull request merged**